examples/twittermap/web/public/javascripts/common/services.js

```diff
@@ -41,7 +41,7 @@ angular.module('cloudberry.common', ['cloudberry.mapresultcache'])
       }
     };
   })
-  .service('cloudberry', function($timeout, cloudberryConfig, MapResultCache) {
+  .service('cloudberry', function($rootScope, $timeout, cloudberryConfig, MapResultCache) {
```

Comment:
Where do we create this $rootScope? Can we put some comments there? What does $rootScope mean...

Reply:
$rootScope is a angularJS thing. It is a way to share information between different controllers and services.

Reply:
I have added some comments about $rootScope

Reply:
Sorry I just found a way to avoid using $rootScope in common/services. We can already use the cloudberry.parameters to share information between different controllers. And the maptype variable can be passed to common/services this way. I will commit a new version.

Reply:
OK. But I can not see anything wrong with put the maptype in $rootScope.

Reply:
I just want to avoid passing $rootScope to common/services. Although I am not sure if there is any performance issue, but the original logic seems to try to avoid that.